pilot/pkg/security/model/authentication.go

```diff
@@ -210,6 +212,76 @@ func AppendURIPrefixToTrustDomain(trustDomainAliases []string) []string {
 	return res
 }
 
+const fips = "fips-140-2"
+
+var fipsCiphers = []string{
+	"ECDHE-ECDSA-AES128-GCM-SHA256",
+	"ECDHE-RSA-AES128-GCM-SHA256",
+	"ECDHE-ECDSA-AES256-GCM-SHA384",
+	"ECDHE-RSA-AES256-GCM-SHA384",
+}
+
+func index(ciphers []string) map[string]struct{} {
+	out := make(map[string]struct{})
+	for _, cipher := range ciphers {
+		out[cipher] = struct{}{}
+	}
+	return out
+}
+
+var fipsCipherIndex = index(fipsCiphers)
+
+// EnforceGoCompliance limits the TLS settings to the compliant values.
+// This should be called as the last policy.
+func EnforceGoCompliance(ctx *gotls.Config, policy string) {
+	switch policy {
+	case "":
+		return
+	case fips:
+		ctx.MinVersion = gotls.VersionTLS12
+		ctx.MaxVersion = gotls.VersionTLS12
+		ctx.CipherSuites = []uint16{
+			gotls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			gotls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			gotls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+			gotls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		}
+		return
+	default:
+		log.Warnf("unknown compliance policy: %q", policy)
+		return
+	}
+}
+
+// EnforceCompliance limits the TLS settings to the compliant values.
+// This should be called as the last policy.
+func EnforceCompliance(ctx *tls.CommonTlsContext, policy string) {
```

Comment:
nit: returning the tls.CommonTlsContext may make usage simpler so they can just wrap the original struct anonymously